standalone/runtime/tests/test_redeem.rs

```diff
@@ -365,124 +365,725 @@ mod spec_based_tests {
+                drop_exchange_rate_and_liquidate(VAULT);
+                assert_noop!(
+                    Call::Redeem(RedeemCall::liquidation_redeem(free_tokens_to_redeem + 1))
+                        .dispatch(origin_of(account_of(ALICE))),
+                    RedeemError::AmountExceedsUserBalance,
+                );
+                let user_tokens_before_redeem = TreasuryPallet::get_free_balance(&account_of(USER));
+                let tokens_to_liquidation_redeem = free_tokens_to_redeem - 10;
+                assert_ok!(Call::Redeem(RedeemCall::liquidation_redeem(free_tokens_to_redeem - 10))
+                    .dispatch(origin_of(account_of(ALICE))));
+                let user_tokens_after_redeem = TreasuryPallet::get_free_balance(&account_of(USER));
 
-    fn set_redeem_period(period: u32) {
-        assert_ok!(Call::Redeem(RedeemCall::set_redeem_period(period)).dispatch(root()));
+                assert_eq!(
+                    user_tokens_before_redeem - tokens_to_liquidation_redeem,
+                    user_tokens_after_redeem
+                )
+            });
+        }
     }
 
-    fn request_redeem() -> H256 {
-        assert_ok!(Call::Redeem(RedeemCall::request_redeem(
-            4_000,
-            BtcAddress::default(),
-            account_of(VAULT)
-        ))
-        .dispatch(origin_of(account_of(USER))));
-        // get the redeem id
-        assert_redeem_request_event()
-    }
+    mod execute_redeem {
+        use redeem::RedeemRequestStatus;
 
-    fn execute_redeem(redeem_id: H256) -> DispatchResultWithPostInfo {
-        ExecuteRedeemBuilder::new(redeem_id).execute()
-    }
+        use super::*;
+        #[test]
+        fn integration_test_redeem_wrapped_execute() {
+            // Checked PRECONDITIONS:
+            // - A pending `RedeemRequest` MUST exist with an id equal to `redeemId`.
+            // - The `rawTx` MUST decode to a valid transaction that transfers exactly the amount specified in the
+            // `RedeemRequest` struct. It MUST be a transaction to the correct address, and provide the expected
+            // OP_RETURN, based on the `RedeemRequest`.
+            // - The `merkleProof` MUST contain a valid proof of of `rawTX`.
+            // - The bitcoin payment MUST have been submitted to the relay chain, and MUST have sufficient
+            //   confirmations.
+            // Checked POSTCONDITIONS:
+            // - The user's `lockedTokens` MUST decrease by `redeemRequest.amountBtc + redeemRequest.transferFeeBtc`.
+            // - The vault’s `toBeRedeemedTokens` MUST decrease by `redeemRequest.amountBtc +
+            //   redeemRequest.transferFeeBtc`.
+            // - The vault’s `issuedTokens` MUST decrease by `redeemRequest.amountBtc + redeemRequest.transferFeeBtc`.
+            // - `redeemRequest.fee` MUST be unlocked and transferred from the redeemer’s account to the fee pool.
+            // - `redeemRequest.status` MUST be set to `Completed`.
```

Comment:
My concern with duplicating the spec is that we have this to sync as well as the implementation, thoughts? cc: @nud3l @sander2 @theodorebugnet

Reply:
Ideally we'd have per-condition URLs which we could link here. That involves some work in the model repo but it will then be easy to keep track of where a condition is tested.

Reply:
Added to backlog: https://www.notion.so/interlay/Add-refs-URLs-to-each-pre-post-condition-in-the-spec-8847d13ac23245eb91e0d76fbab8888d